src/react/components/common/assetPreview/assetPreview.tsx

```diff
@@ -85,38 +86,40 @@ export class AssetPreview extends React.Component<IAssetPreviewProps, IAssetPrev
         return (
             <div className={classNames.join(" ")}>
                 <div className="asset-preview-container">
-                    {!loaded &&
+                    {!loaded && !asset.hasError &&
                         <div className="asset-loading">
                             <i className="fas fa-circle-notch fa-spin" />
                         </div>
                     }
-                    {asset.type === AssetType.Image &&
-                        <ImageAsset asset={rootAsset}
-                            additionalSettings={this.props.additionalSettings}
-                            onLoaded={this.onAssetLoad}
-                            onActivated={this.props.onActivated}
-                            onDeactivated={this.props.onDeactivated} />
-                    }
-                    {(asset.type === AssetType.Video || asset.type === AssetType.VideoFrame) &&
-                        <VideoAsset asset={rootAsset}
-                            additionalSettings={this.props.additionalSettings}
-                            childAssets={childAssets}
-                            timestamp={asset.timestamp}
-                            autoPlay={autoPlay}
-                            onLoaded={this.onAssetLoad}
-                            onChildAssetSelected={this.props.onChildAssetSelected}
-                            onActivated={this.props.onActivated}
-                            onDeactivated={this.props.onDeactivated} />
-                    }
-                    {asset.type === AssetType.TFRecord &&
-                        <TFRecordAsset asset={asset}
-                            onLoaded={this.onAssetLoad}
-                            onActivated={this.props.onActivated}
-                            onDeactivated={this.props.onDeactivated} />
-                    }
-                    {asset.type === AssetType.Unknown &&
-                        <div className="asset-error">{strings.editorPage.assetError}</div>
-                    }
+                    {!asset.hasError ?
```

Comment:
Let's invert this expression, if there is an asset error then show the new experience and return.  Consider breaking down the render function into smaller render methods.